lightning/src/sign/mod.rs

```diff
@@ -947,14 +947,20 @@ impl InMemorySigner {
 		if spend_tx.input[input_idx].previous_output != descriptor.outpoint.into_bitcoin_outpoint() { return Err(()); }
 
 		let remotepubkey = bitcoin::PublicKey::new(self.pubkeys().payment_point);
-		let witness_script = if self.channel_type_features().supports_anchors_zero_fee_htlc_tx() {
+		// We cannot always assume that `channel_parameters` is set, so can't just call
+		// `self.channel_parameters()` or anything that relies on it
+		let supports_anchors_zero_fee_htlc_tx = self.channel_parameters.as_ref()
+			.map(|params| params.channel_type_features.supports_anchors_zero_fee_htlc_tx())
+			.unwrap_or(false);
```

Comment:
Might be worth adding a helper. We do this in `create_spendable_outputs_psbt`, too, and it looks like we have an unchecked use in `sign_counterparty_commitment`. Though maybe that is never reached in practice?

Reply:
Turns out the one in `create_spendable_outputs_psbt` is not on `self` but rather on `StaticPaymentOutputDescriptor`. While we could DRY up that usage and `StaticPaymentOutputDescriptor::max_witness_length`, it's probably not worth it because we may refactor the former as per https://github.com/lightningdevkit/rust-lightning/pull/2605#discussion_r1341709819.